pkg/services/dashboards/dashboard_service_mock.go

```diff
@@ -327,17 +329,26 @@ func (_m *FakeDashboardService) SaveDashboard(ctx context.Context, dto *SaveDash
 }
 
 // SearchDashboards provides a mock function with given fields: ctx, query
-func (_m *FakeDashboardService) SearchDashboards(ctx context.Context, query *FindPersistedDashboardsQuery) error {
+func (_m *FakeDashboardService) SearchDashboards(ctx context.Context, query *FindPersistedDashboardsQuery) (model.HitList, error) {
 	ret := _m.Called(ctx, query)
 
-	var r0 error
-	if rf, ok := ret.Get(0).(func(context.Context, *FindPersistedDashboardsQuery) error); ok {
+	var r0 model.HitList
+	if rf, ok := ret.Get(0).(func(context.Context, *FindPersistedDashboardsQuery) model.HitList); ok {
```

Comment:
I'm a bit confused about how the mocking here works. This is a more general educational question for me. According to the docs `Get Returns the argument at the specified index.` Where do the arguments get set and how come in this case it could either be a function or the return value of the function? So it could either be of type `func(context.Context, *FindPersistedDashboardsQuery) model.HitList` or `model.HitList`.